checkpoint.go

```diff
@@ -118,27 +117,15 @@ func Checkpoint(w *wal.WAL, from, to int, keep func(id uint64) bool, mint int64)
 			// Ignore WAL files below the checkpoint. They shouldn't exist to begin with.
 			from = last
 
-			r, err := wal.NewSegmentsReader(filepath.Join(w.Dir(), dir))
-			if err != nil {
-				return nil, errors.Wrap(err, "open last checkpoint")
-			}
-			defer r.Close()
-			closers = append(closers, r)
```

Comment:
@gouthamve 	do you remember why using just `defer r.Close()` is not enough?

Reply:
ping @gouthamve

Reply:
yeah I read this one few times , but you are calling `closeAll(closers...)` right at the end before the return so it is essentially the same as defer.

Reply:
Ah, I see :) When I first added this, it had deletion of the segments in the function itself :P Now they have been moved out of the function.

See: https://github.com/prometheus/tsdb/pull/392/files#diff-682f9cc560724652272cca157347157bR275

Reply:
ah nice, so this PR is correct?